test/NuGet.Core.Tests/NuGet.Packaging.Test/LicensesTests/LicenseExpressionTokenizerTests.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class LicenseExpressionTokenizerTests
+    {
+        [Theory]
+        [InlineData("")]
+        [InlineData("  ")]
+        [InlineData(null)]
+
+        public void LicenseExpressionTokenizer_ThrowsForInvalidLicenseStrings(string value)
+        {
+            Assert.Throws<ArgumentNullException>(() => new LicenseExpressionTokenizer(value));
+        }
+
+        [Theory]
+        [InlineData("MIT")]
+        [InlineData("value with invalid characters _ * ! ? :)")]
+        public void LicenseExpressionTokenizer_DoesNotThrowForNonEmptyStrings(string value)
+        {
+            var tokenizer = new LicenseExpressionTokenizer(value);
+        }
+
+        [Theory]
+        [InlineData("MIT", "VALUE")]
+        [InlineData("(", "OPENING_BRACKET")]
+        [InlineData(")", "CLOSING_BRACKET")]
+        [InlineData("AND", "AND")]
+        [InlineData("OR", "OR")]
+        [InlineData("WITH", "WITH")]
+        [InlineData("with", "VALUE")]
+        [InlineData("Or", "VALUE")]
+        [InlineData("aND", "VALUE")]
+        public void LicenseExpressionTokenizer_TokenizesSimpleExpressionCorrectly(string value, string type)
+        {
+            Enum.TryParse(type, out LicenseTokenType tokenType);
```

Comment:
Why not have `tokenType` as a method argument instead of string parsing?

Reply:
Because the type is internal (exposed for tests via internalsvisibleto). However xunit needs the parameter types to be public.